### ISSUE TITLE
CI Feature: Add a package.json linter to ensure static versions are used

### ISSUE THREAD

**Issue by ardelato** [iFixit/react-commerce#1346]
## Description

There are times when we install a new package but the semantic version syntax may default to pre-pending `^` or `~`. The rest of our packages all use a static semantic version.

It would be nice if we could automate this or at least catch it so the semantic version syntax is consistent across our `package.json`.

**References:**
- https://github.com/iFixit/react-commerce/pull/1216#discussion_r1062787286
- https://github.com/iFixit/react-commerce/pull/1319#discussion_r1095063250

**Comment by mlahargou** [comment]
Taking this off of QAE backlog as it looks like IoP is going to take care of this.

**Comment by federicobadini** [comment]
Probably the best option we have is using `npm-package-json-lint` adding it to the build & lint step in order for it to error out if some rule does not pass validation.
It just requires a base config like this:

```json
{
  "rules": {
    "prefer-absolute-version-dependencies": ["error", {}]
  }
}

```

We should also make sure to skip packages that have wildcard - like our own `@ifixit/*` packages - from validation, but it should be feasible too.

There are more lightweight alternatives like adding `"save-exact": "true"` to package.json or .npmrc.
However these solutions only reduce the likelyhood of non-static package version making their way into package.json.
So if the requirement is strict, I'd go with the first one.

Let me know what you think

**Comment by ardelato** [comment]
I tested the `save-exact` option which is useful for adding new packages as the developers won't need to specify a version every time they want to install a package. https://github.com/iFixit/react-commerce/compare/testing-package-json-linter

However, to be more strict I tested the `npm-package-json-lint` package and found out the following.
1. The `exception` setting does not take in `regex` and therefore we need to explicitly list all modules we want to exclude. This means we need to list each of the `@ifixit` packages.
2. If we make use of the `lint-staged` `pre-commit` hook we already have, then we can run this linter every time we commit. 

---

I think we should include both the `save-exact` rule as well as the `npm-package-json-lint` package. Although, how we integrate the `npm-package-json-lint` package is still up for discussion.

If we use the `pre-commit` hook then it would be included as part of the normal workflow of a Dev. 

If we use it in CI, then we would need to add an additional job as the current [`linter` action ](https://github.com/iFixit/react-commerce/blob/main/.github/workflows/lint.yml#L29) only runs `eslint` for the `frontend` directory and will automatically commit the `eslint` fixes.
